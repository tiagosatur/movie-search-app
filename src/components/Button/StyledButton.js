import styled, { css } from 'styled-components';
import { colorPallete, rem } from '../../style';

const buttonBase = css`
  align-items: center;
  background-color: transparent;
  border-color: transparent;
  border-radius: ${rem(3)};
  border-style: solid;
  border-width: ${rem(1)};
  color: ${colorPallete.black};
  cursor: pointer;
  font-size: ${rem(14)};
  font-weight: bold;
  display: inline-flex;
  justify-content: center;
  padding: ${rem(8)} ${rem(16)};
  position: relative;
  transition: all .5s;
`;

export const StyledButton = styled.button`
  ${buttonBase}
  
  color: ${({ solid, outlined, text, anchor }) =>
    solid
      ? colorPallete.white
      : text
      ? colorPallete.black
      : outlined || anchor
      ? colorPallete.primaryGradient
      : colorPallete.black};

  background: ${({ solid, text, anchor, outlined }) =>
    solid
      ? colorPallete.primaryGradient
      : outlined || anchor || text
      ? colorPallete.transparent
      : colorPallete.transparent};
  
  
  border-radius: ${({ rounded }) => (rounded ? '50%' : rem(4))};

  border-bottom-right-radius: ${({ rounded, left }) =>
    rounded && left ? '0' : rem(4)};
  
  border-top-right-radius: ${(rounded, left) =>
    rounded && left ? '0' : rem(4)};

  border-top-left-radius: ${({ rounded, right }) =>
    rounded && right ? '0' : rem(4)};

  border-bottom-left-radius: ${({ rounded, right }) =>
    rounded && right ? '0' : rem(4)};
  
  height: ${({ rounded, sm, md, lg }) =>
    rounded && sm
      ? rem(30)
      : rounded && md
      ? rem(40)
      : rounded && lg
      ? rem(50)
      : `auto`};

  padding: ${({ rounded }) => (rounded ? rem(8) : rem(8) + ' ' + rem(16))};

  a {
    color: ${({ text, anchor }) =>
      text
        ? colorPallete.black
        : anchor
        ? colorPallete.primary
        : colorPallete.black};
  }
`;
