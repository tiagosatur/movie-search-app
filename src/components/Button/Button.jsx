
import React from 'react';
import { StyledButton } from './StyledButton';

const Button = ({
  type,
  handleClick,
  classes,
  iconName,
  children,
  disabled,
  title,
  ...styledProps
}) => {

  const defineType = (!type || type == '') ? 'submit': type;
  
  return (
    <StyledButton
      type={defineType}
      onClick={handleClick}
      className={`button ${classes ? classes : ''}`}
      disabled={disabled}
      title={title ? title : ''}
      {...styledProps}
    >
      {iconName ? <i className={iconName} /> : null}
      {children}
    </StyledButton>
  );
};

export default Button;
