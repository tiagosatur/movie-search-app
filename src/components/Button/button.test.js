import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './Button';
import { stub } from 'sinon';

const onButtonClick = stub();

describe('Button tests', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Button solid handleClick={onButtonClick} />
    ).dive();
  });
  
  it('should render the Button element', () => {
    expect(wrapper.find('.button').exists()).toBe(true);
  });


  /* it('simulates click events', () => {
    wrapper.find('.button').prop('onClick')();
    expect(onButtonClick.callCount).to.equal(1)
    
  }); */

  it('should call Button snapshot correctly', () => {
    const tree = renderer
        .create(wrapper)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });  

})