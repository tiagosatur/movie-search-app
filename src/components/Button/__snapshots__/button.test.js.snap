// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should call Button snapshot correctly 1`] = `
<button
  className="button  sc-bdVaJa jllYbe"
  onClick={[Function]}
  title=""
  type="submit"
/>
`;
